pylint/checkers/base/basic_checker.py

```diff
@@ -590,8 +595,72 @@ def _check_misplaced_format_function(self, call_node):
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    def _check_builtins(self, node: nodes.Call) -> None:
+        """Visit a Call node -> check it's a built-in and the number of parameters is correct."""
+        # TODO: add checks for other built-ins
+        if node.func.name in ("aiter", "all", "any"):
+            # These built-ins all take exactly one argument, and it must be iterable
+            # TODO: check the argument is iterable
+            if len(node.args) != 1:
+                self.add_message(
+                    "incorrect-number-of-parameters", node=node, args=(node.func.name,)
+                )
+        if node.func.name in (
+            "abs",
+            "ascii",
+            "bin",
+            "callable",
+            "chr",
+            "hex",
+            "id",
+            "len",
+            "vars",
+        ):
+            # These built-ins all take exactly one argument, which can be not iterable
+            if len(node.args) != 1:
+                self.add_message(
+                    "incorrect-number-of-parameters", node=node, args=(node.func.name,)
+                )
+        elif node.func.name in (
+            "bool",
+            "dir",
+            "float",
+            "tuple",
+            "list",
+            "hash",
+            "input",
+            "memoryview",
+            "oct",
+            "ord",
+            "repr",
+            "reversed",
+            "set",
+        ):
+            # These built-ins take 0 or 1 argument
+            if len(node.args) > 1:
+                self.add_message(
+                    "incorrect-number-of-parameters", node=node, args=(node.func.name,)
+                )
+        elif node.func.name in (
+            "delattr",
+            "divmod",
+            "filter",
+            "hasattr",
+            "help",
+            "isinstance",
+        ):
+            # These built-ins take 2 arguments
+            if len(node.args) != 2:
+                self.add_message(
+                    "incorrect-number-of-parameters", node=node, args=(node.func.name,)
```

Comment:
Please add confidence=HIGH

Reply:
added in commit [c7f3753](https://github.com/PyCQA/pylint/pull/6280/commits/c7f3753b78313726a9b3ac7784534ff3ad1dd92a)